### ISSUE TITLE
RStudio Desktop on Windows: wrong encoding in popup help during code completion

### ISSUE THREAD

**Issue by vh-d** [rstudio/rstudio#1995]
I have a package with man pages with text in central European language encoded in UTF-8 (the package is not intended for public). 

Generated html man pages has charset declared correctly and RStudio help pane renders the man pages correctly.

![scr1](https://user-images.githubusercontent.com/8167061/35047745-2158196a-fb9b-11e7-8b9f-d97aecffa7b2.GIF)

However, when hovering mouse over code-completed function name, RStudio code shows header of man page of the function. This short part of the man page is now incorrectly interpreted as ASCII.

![scr2](https://user-images.githubusercontent.com/8167061/35047761-3301a154-fb9b-11e7-8985-1e8438ef0ab3.GIF)

I am not on the latest build

```
> RStudio.Version()

...

$mode
[1] "desktop"

$version
[1] ‘1.1.393’
```

but I haven't found any commit mentioning fixing encoding issue.

```
R version 3.4.3 (2017-11-30)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 7 x64 (build 7601) Service Pack 1

Matrix products: default

locale:
  [1] LC_COLLATE=Czech_Czech Republic.1250  LC_CTYPE=Czech_Czech Republic.1250    LC_MONETARY=Czech_Czech Republic.1250
[4] LC_NUMERIC=C                          LC_TIME=Czech_Czech Republic.1250    

attached base packages:
  [1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
  [1] encbug_0.1.0

loaded via a namespace (and not attached):
  [1] compiler_3.4.3 tools_3.4.3    yaml_2.1.16   

```

**Comment by kevinushey** [comment]
Does it make a difference if you add the line

    Encoding: UTF-8

to your `DESCRIPTION` file? (It would also be helpful if you could share the sources to your `encbug` package just so we can try to reproduce locally)

**Comment by vh-d** [comment]
I created the package by simply
1. starting a new RStudio project -> package
1. deleting the original hello.Rd and NAMESPACE files
1. putting `#' @export` and some text with non-latin letters above the hello world function
1. building documentation using roxygen2
1. building the package

RStudio or roxygen have added `Encoding: UTF-8` into DESCRIPTION automatically for me.

**Comment by vh-d** [comment]
Also, I have just tested the same example on Linux and the text is displayed correctly (as UTF-8) so it seems limited to Windows platform.

I also tried changing locale settings on a Windows machine (so that Sys.getlocale() returns different results) and the interpretation of non-latin characters changes so it seems RStudio assumes the text is encoded in the native encoding I guess.

**Comment by mrchypark** [comment]
I have an almost same issue but I got an error.

![image](https://user-images.githubusercontent.com/6179259/39884025-7f6f708a-54c3-11e8-88ed-71dc52b4b79c.png)

I found same my issue in Japanese.

https://ja.stackoverflow.com/questions/24083/rstudio%E3%81%A7%E3%81%AErmecab%E3%81%AE%E8%B5%B7%E5%8B%95%E6%99%82%E3%81%AE%E3%82%A8%E3%83%A9%E3%83%BC%E3%83%A1%E3%83%83%E3%82%BB%E3%83%BC%E3%82%B8

**Comment by mrchypark** [comment]
> sessionInfo()
R version 3.4.4 (2018-03-15)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows >= 8 x64 (build 9200)

Matrix products: default

locale:
[1] LC_COLLATE=Korean_Korea.949  LC_CTYPE=Korean_Korea.949    LC_MONETARY=Korean_Korea.949
[4] LC_NUMERIC=C                 LC_TIME=Korean_Korea.949    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] RevoUtils_10.0.9     RevoUtilsMath_10.0.1

loaded via a namespace (and not attached):
 [1] presidentSpeechKr_0.1.1 httr_1.3.1              compiler_3.4.4          magrittr_1.5           
 [5] R6_2.2.2                tools_3.4.4             pillar_1.2.1            tibble_1.4.2           
 [9] yaml_2.1.18             Rcpp_0.12.16            xml2_1.2.0              rvest_0.3.2            
[13] rlang_0.2.0

**Comment by mrchypark** [comment]
Does anyone care about this issue?

**Comment by kevinushey** [comment]
Sorry -- we do care about this issue, but our time and resources are limited and we haven't yet been able to acquire the time to diagnose and fix this issue.

**Comment by mrchypark** [comment]
@kevinushey I'm not want to rush you guys. Sorry about that. It's good news to add v1.3 milestone. I'll wait for it. Thank you all your effort.

**Comment by dpprdan** [comment]
The second screenshot in the original post looks like a UTF-8 string marked as native encoding (cp1250). I.e. `Encoding(x) <- "UTF-8"` is missing.

**Comment by mikebessuille** [comment]
Kevin asking that we re-test this bug for 1.4 based on other encoding changes that may have fixed this.

**Comment by vh-d** [comment]
I have just installed RStudio daily build 1.4.626 and... yeah, something's changed for sure!

![rstudio_1 4_screenshot](https://user-images.githubusercontent.com/8167061/88217648-256bc280-cc5f-11ea-9d5f-40ca817079e4.png)

Everything is upside-down, all the text is gibberish. Its fun.

**Comment by mikebessuille** [comment]
The upside-down UI, while immensely fun for a short time, has been seen by others across several builds.  https://github.com/rstudio/rstudio/issues/7345.  It seems unrelated to the original issue.  See the resolution ideas in #7345 to resolve the upside down UI problem and let us know if that fixes it.

**Comment by vh-d** [comment]
I have solved the rendering problem with my build (as in #7356), but the encoding problems in tooltips remain.

**Comment by mikebessuille** [comment]
I assume you mean #7345?  What did you do to resolve the upside-down UI problem?

**Comment by vh-d** [comment]
> I assume you mean #7345? What did you do to resolve the upside-down UI problem?

Yes, I mean #7345. I have switched RStudio to software rendering. I run Windows in VirtualBox as the OP.

**Comment by kevinushey** [comment]
Looks like this is fixed now in the latest daily builds. Using the example package at https://github.com/kevinushey/nihao, installed via:

```
renv::install("kevinushey/nihao")
library(nihao)
```

I see:

<img width="596" alt="Screen Shot 2020-10-15 at 2 57 42 PM" src="https://user-images.githubusercontent.com/1976582/96190257-c8e31100-0ef6-11eb-9af0-8d9814f47328.png">

If you're still having issues, please feel free to re-open. Thanks!

**Comment by kevinushey** [comment]
@melissa-barca that's expected -- the first case is showing the help topic for the `nihao` package itself; the second is showing the help topic for the `nihao` function from the `nihao` package. (Sorry for making the example slightly confusing here!)

**Comment by melissa-barca** [comment]
Verified in 1.4.954 on Windows and MacOS.

**Comment by mrchypark** [comment]
In my case, solved in 1.4.953 on windows 10 19042.608 with Korean language.
